control/grpc.py

```diff
@@ -44,6 +68,7 @@ def __init__(self, config, gateway_state, spdk_rpc_client) -> None:
             self.logger.info(f"Using NVMeoF gateway version {ver}")
         self.config = config
         self.logger.info(f"Using configuration file {config.filepath}")
+        self.rpc_lock = GatewayService.RPCGuard(self.logger)
```

Comment:
```suggestion
        self.rpc_lock = threading.Lock()
```

Additionally class RPCGuard could be removed

Reply:
Removed